cargo-dylint/tests/ci.rs

```diff
@@ -133,6 +133,165 @@ fn cargo_dylint_and_dylint_readmes_are_equal() {
+    let link_re = Regex::new(r"\[`([^`]+)`\]\([^)]+\)").unwrap();
+    let desc_re = Regex::new(r"\|[^|]*\| ([^|]+) \|").unwrap();
+    
+    let mut result = Vec::new();
+    let mut lines = s.lines().peekable();
+    
+    while let Some(line) = lines.next() {
+        if let (Some(link_cap), Some(desc_cap)) = (link_re.captures(line), desc_re.captures(line)) {
+            result.push((
+                link_cap.get(1).unwrap().as_str().to_string(),
+                desc_cap.get(1).unwrap().as_str().trim().to_string()
+            ));
+        }
+    }
+    
+    result.sort();
+    result
+}
+
+fn generate_lint_tables(examples_dir: &Path, categories: &[&str]) -> String {
+    let mut content = String::new();
+    
+    // Generate the tables for each category
+    for category in categories {
+        use std::fmt::Write;
+        write!(content, "\n## {}\n\n", capitalize(category)).unwrap();
+        content.push_str("| Example | Description/check |\n");
+        content.push_str("| - | - |\n");
+        
+        // Get the examples for this category
+        let examples = collect_examples_from_category(examples_dir, category);
+        for (name, description) in examples {
+            writeln!(content, "| [`{name}`](./{category}/{name}) | {description} |").unwrap();
+        }
+    }
+    
+    content
+}
+
+fn extract_between_markers(content: &str, start_marker: &str, end_marker: &str) -> Option<String> {
+    let start = content.find(start_marker)? + start_marker.len();
+    let end = content.find(end_marker)?;
+    Some(content[start..end].trim().to_string())
+}
+
+fn collect_examples_from_category(examples_dir: &Path, category: &str) -> Vec<(String, String)> {
+    let mut examples = Vec::new();
+    let category_dir = examples_dir.join(category);
+    
+    for entry in read_dir(&category_dir).unwrap() {
+        let entry = entry.unwrap();
+        let metadata = entry.metadata().unwrap();
+        if metadata.is_dir() {
+            let cargo_path = entry.path().join("Cargo.toml");
+            if cargo_path.exists() {
+                if let Some((name, desc)) = extract_name_and_description(&cargo_path) {
+                    examples.push((name, desc));
+                }
+            }
+        }
+    }
+    
+    // Sort examples by name
+    examples.sort_by(|(a_0, _), (b_0, _)| a_0.cmp(b_0));
+    examples
+}
+
+fn extract_name_and_description(cargo_path: &Path) -> Option<(String, String)> {
+    let content = read_to_string(cargo_path).ok()?;
+    
+    // Get the name from the directory
+    let name = cargo_path
+        .parent()
+        .and_then(|path| path.file_name())
+        .unwrap()
+        .to_string_lossy()
+        .to_string();
+    
+    // Extract the description using regex
+    let re = Regex::new(r#"description\s*=\s*"([^"]*)""#).unwrap();
+    let description = if let Some(caps) = re.captures(&content) {
+        if let Some(desc) = caps.get(1) {
```

Comment:
```suggestion
        let desc = caps.get(1).unwrap();
```
If the `captures` call succeeds, then this call should succeed as well. 

A failure of this `get` would likely indicate a bug in the definition of `re`, which I think is worth a test panic.

Note that removing this `if` will require other changes.